packages/components/organisms/f-header/test-utils/component-objects/f-header.component.js

```diff
@@ -1,66 +1,140 @@
-import {
+const Page = require('@justeat/f-wdio-utils/src/page.object');
+const {
     HEADER_COMPONENT, 
     HEADER_LOGO, 
     MOBILE_NAVIGATION_BAR, 
     NAVIGATION
-} from './f-header.selectors';
-
-// Component
-const headerComponent = () => $(HEADER_COMPONENT);
-
-// Logo
-const headerLogo = () => $(HEADER_LOGO);
-
-// Navigation
-const navigation = {
-    offers: {
-        link: () => $(NAVIGATION.offers.link), 
-        icon: () => $$(NAVIGATION.offers.link)
-    }, 
-    help: {
-        link: () => $(NAVIGATION.help.link), 
-    }, 
-    delivery: {
-        link: () => $(NAVIGATION.delivery.link), 
-    }, 
-    userAccount: {
-        link: () => $(NAVIGATION.userAccount.link), 
-    } 
-}
-const mobileNavigationBar = () => $(MOBILE_NAVIGATION_BAR);
-const mobileOffersIcon = () => 
-    navigation.offers.icon().filter(element => element.getAttribute('class').includes('u-showBelowMid'));
-    
-const webOffersIcon = () =>
-    navigation.offers.icon().filter(element => element.getAttribute('class').includes('u-showAboveMid'));
+} = require ('./f-header.selectors');
 
-// Functions
+class Header extends Page {
 
-exports.waitForHeader = () => headerComponent().waitForExist();
-exports.isFieldLinkDisplayed = fieldName => navigation[fieldName].link().isDisplayedInViewport();
-exports.isLogoDisplayed = () => headerLogo().isDisplayedInViewport();
-exports.isMobileNavigationBarVisible = () => mobileNavigationBar().isDisplayedInViewport();
+    get component () { return $(HEADER_COMPONENT) }
+    get logo () { return  $(HEADER_LOGO) }
+    get mobileNavigationBar () { return $(MOBILE_NAVIGATION_BAR) }
 
-exports.isMobileOffersIconDisplayed = () => {
-    const element = mobileOffersIcon();
+    navigation = {
+        offers: {
+            get link () { return $(NAVIGATION.offers.link) }
+        }, 
+        help: {
+            get link () { return $(NAVIGATION.help.link) }
+        }, 
+        delivery: {
+            get link () { return $(NAVIGATION.delivery.link) }, 
+        }, 
+        userAccount: {
+            get link () { return $(NAVIGATION.userAccount.link) }
+        }, 
+        countrySelector: {
+            get link () { return  $(NAVIGATION.countrySelector.link) },
+            get currentIcon () { return $(NAVIGATION.countrySelector.currentIcon) },
+            get countries() { return $$(NAVIGATION.countrySelector.countryList) } 
+        }
+    }
 
-    return element.length === 1 && element[0].isDisplayedInViewport();
-};
+    get countryLink()  { return this.countryValue != null ?  this.countryValue : 'Please set a country value'; }
```

Comment:
Nit: Should probably be countryListItemLink

Reply:
Nit: Should probably through an error here for the else condition, rather than return a string of 'Please set a country value'